--- a/runtime/src/iree/hal/utils/deferred_command_buffer.c
+++ b/runtime/src/iree/hal/utils/deferred_command_buffer.c
@@ -720,7 +720,8 @@
       sizeof(iree_hal_cmd_dispatch_t), &total_size,
       IREE_STRUCT_FIELD_ALIGNED(constants.data_length, uint8_t,
                                 iree_max_align_t, &constants_offset),
-      IREE_STRUCT_FIELD_ALIGNED(bindings.count, iree_hal_buffer_ref_t, 1,
+      IREE_STRUCT_FIELD_ALIGNED(bindings.count, iree_hal_buffer_ref_t,
+                                iree_alignof(iree_hal_buffer_ref_t),
                                 &bindings_offset)));
 
   iree_hal_cmd_dispatch_t* cmd = NULL;
